### PR TITLE
tests: add a functional test for upgrade

### PR DESCRIPTION
**Pull request by phlogistonjohn**
While this probably needs to get fleshed out more over time this seems
like a pretty decent start for a test of upgrading the db across
versions of Heketi.

### CODE REVIEW

File: tests/functional/TestUpgrade/heketi.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    "_db_comment": "Database file name",
+    "db": "heketi.db",
+
+    "_loglevel_comment": [
+      "Set log level. Choices are:",
+      "  none, critical, error, warning, info, debug",
+      "Default is warning"
+    ],
+    "loglevel" : "bladebug",
```
Comment (raghavendra-talur) [inline tests/functional/TestUpgrade/heketi.json]: typo?

File: tests/functional/TestUpgrade/run.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        exit 0
+    fi
+
+    rm -rf .env
+    virtualenv .env
+    . .env/bin/activate
+    pip install -r "$HEKETI_DIR/client/api/python/requirements.txt"
+
+    echo '----> Running test_upgrade.py'
+    exec python test_upgrade.py "$@" -v
```
Comment (raghavendra-talur) [inline tests/functional/TestUpgrade/run.sh]: Why does this invocation have a "-v" flag and not the one below it?

File: tests/functional/TestUpgrade/test_upgrade.py [hunk 3]
```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python
+#
+# Copyright (c) 2018 The heketi Authors
+#
+# This file is licensed to you under your choice of the GNU Lesser
+# General Public License, version 3 or any later version (LGPLv3 or
+# later), as published by the Free Software Foundation,
```
Comment (raghavendra-talur) [inline tests/functional/TestUpgrade/test_upgrade.py]: I guess you copied the license part from python api file. Heketi license is currentyl on GPL 3 or LGPLv3 for core code( api server, tests etc). Only the API is dual licensed under Apache.

**Comment by raghavendra-talur** [comment]
@phlogistonjohn Congratulations for getting PR #1000 :) 👍 

@humblec @obnoxxx @purpleidea ^^

**Comment by humblec** [comment]
Congrats @phlogistonjohn

File: tests/functional/TestUpgrade/heketi.json (hunk 1 above)
Comment (phlogistonjohn) [reply tests/functional/TestUpgrade/heketi.json]: That is odd. I don't remember making that change. :-\   Will fix.

File: tests/functional/TestUpgrade/run.sh (hunk 2 above)
Comment (phlogistonjohn) [reply tests/functional/TestUpgrade/run.sh]: Good point. I will make them consistent.

File: tests/functional/TestUpgrade/test_upgrade.py (hunk 3 above)
Comment (phlogistonjohn) [reply tests/functional/TestUpgrade/test_upgrade.py]: Yes, that is exactly what I did. What do you think it should be?

**Comment by phlogistonjohn** [comment]
> @phlogistonjohn Congratulations for getting PR #1000 :)

Hah. I wasn't even trying. :-D

**Comment by phlogistonjohn** [comment]
Review comments addressed.

File: tests/functional/TestUpgrade/test_upgrade.py (hunk 3 above)
Comment (obnoxxx) [reply tests/functional/TestUpgrade/test_upgrade.py]: As in any non-api go file, e.g:

> // This file is licensed to you under your choice of the GNU Lesser
> // General Public License, version 3 or any later version (LGPLv3 or
> // later), or the GNU General Public License, version 2 (GPLv2), in all
> // cases as published by the Free Software Foundation.

File: tests/functional/TestUpgrade/run.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+    rm -rf .env
+    virtualenv .env
+    . .env/bin/activate
+    pip install -r "$HEKETI_DIR/client/api/python/requirements.txt"
+
+    echo '----> Running test_upgrade.py'
+    exec python test_upgrade.py -v "$@"
+else
+    exec python test_upgrade.py -v "$@"
```
Comment (obnoxxx) [inline tests/functional/TestUpgrade/run.sh]: now these exec statements are exactly the same, why don't you put them as a single statement after the if block (and omit the else)?

Comment (phlogistonjohn) [reply tests/functional/TestUpgrade/run.sh]: Taken care of (license should have already been updated).

**Comment by obnoxxx** [comment]
LGTM, thanks!
Let's wait for the centos-ci run to pass once...

**Comment by obnoxxx** [comment]
@raghavendra-talur please confirm all your requests have been addressed...

**Comment by obnoxxx** [comment]
vm type issue again... 

retest this please

**Pull request merged**